### DISCUSSION TITLE
怎么正确运行本地测试命令

### DISCUSSION THREAD

**Discussion by yangxiuxiu1115** [ant-design/ant-design#42585] [Q&A]
> 如题所示，我正在尝试解决一些简单问题，但是在运行测试用例过程中我遇到一些问题无法正确通过测试用例（我并未修改代码）
> 我已尽我所能去寻找文档，可能是由于我自身的疏忽并未找到对应文档。希望各位能就以下这几个例子给与一些指导，感激不尽。
* rc-select、rc-tree等这些rc库的测试用例
* antd本身的测试用例

**Comment by MadCcc** [comment]
关注一下目标分支的 CI，如果 CI 测试是通过的，多半是本地依赖有问题，删除 lock 文件后重新 install 即可

**Reply by yangxiuxiu1115** [reply]
那我如果想要在本地跑测试命令呢。

**Comment by kiner-tang** [comment]
你可以先看看这个博客是否对你的问题有所帮助：https://ant.design/docs/blog/contributor-development-maintenance-guide-cn#%E5%BF%AB%E7%85%A7%E6%9B%B4%E6%96%B0%E9%97%AE%E9%A2%98

**Comment by BoyYangzai** [comment]
1. rc-component 可以直接 npm run test
2. antd 参考 https://ant.design/docs/react/contributing-cn#pull-request

**Reply by yangxiuxiu1115** [reply]
我在rc-component中直接运行了 npm run test，但是结果好像不尽人意。
如下图所示 
![image](https://github.com/ant-design/ant-design/assets/79584569/41186627-b7b8-42d3-b12d-05012679be76)

**Reply by BoyYangzai** [reply]
> 我在rc-component中直接运行了 npm run test，但是结果好像不尽人意。 如下图所示 ![image](https://user-images.githubusercontent.com/79584569/240631775-41186627-b7b8-42d3-b12d-05012679be76.png)

结果表示你的代码没有通过测试诶
1.大部分情况是你的修改导致组件功能发生变化 所以无法通过测试 所以需要继续优化更改 保证功能完全通过 test
2.如果你真的确定是测试用例有误 可以联系antd维护者进行处理 当然 百分之99的情况都是第一种

**Reply by RedJue** [reply]
建议先同步上游代码，再删掉 node_modules 重新安装试试，有时候依赖更新也会导致跑用例失败，当然这是在你代码没问题的前提下。

**Reply by yangxiuxiu1115** [reply]
我并没有修改代码，我直接拉下来的最新代码 。

**Reply by yangxiuxiu1115** [reply]
> 建议先同步上游代码，再删掉 node_modules 重新安装试试，有时候依赖更新也会导致跑用例失败。

好的，我再尝试以下。

**Reply by BoyYangzai** [reply]
> 我并没有修改代码，我直接拉下来的最新代码 。

@RedJue 果汁哥说的对 如果不是你的修改原因 那记得检查是不是依赖没有重新安装更新

**Reply by yangxiuxiu1115** [reply]
或许因为windows的原因吗。

**Reply by yangxiuxiu1115** [reply]
我尝试的拉取新的代码，重新安装了依赖。依旧出现了问题，不能正确通过测试。
![image](https://github.com/ant-design/ant-design/assets/79584569/e0906731-d960-44dd-a523-e84c72789e55)

**Reply by RedJue** [reply]
> 或许因为windows的原因吗。

我看截图里你用了pnpm，建议用yarn 或者 cnpm，pnpm 对antd对兼容性不是很好，可以尝试一下。

**Reply by yangxiuxiu1115** [reply]
好的，我尝试使用npm。

**Reply by BoyYangzai** [reply]
( 建议使用mac🤪

**Reply by yangxiuxiu1115** [reply]
npm通过了用例，感谢！我即将关闭这个discussions

**Reply by BoyYangzai** [reply]
🎉

**Comment by yangxiuxiu1115** [comment]
这是我的基础环境信息。
![image](https://github.com/ant-design/ant-design/assets/79584569/4a40614f-f75c-4123-963b-67863bc5245f)
